source/shared/cpp/ObjectModel/MarkDownParsedResult.cpp

```diff
@@ -0,0 +1,193 @@
+#include "MarkDownParsedResult.h"
+
+using namespace AdaptiveCards;
+
+void MarkDownParsedResult::Translate()
+{
+    MatchLeftAndRightEmphasises();
+}
+
+void MarkDownParsedResult::Append(MarkDownParsedResult &x)
+{
+    m_codeGenTokens.splice(m_codeGenTokens.end(), x.m_codeGenTokens);
+    m_emphasisLookUpTable.splice(m_emphasisLookUpTable.end(), x.m_emphasisLookUpTable);
+}
+
+void MarkDownParsedResult::Append(std::shared_ptr<MarkDownHtmlGenerator> &x) 
+{
+    m_codeGenTokens.push_back(x);
+}
+
+void MarkDownParsedResult::Append(std::shared_ptr<MarkDownEmphasisHtmlGenerator> &x) 
+{
+    m_emphasisLookUpTable.push_back(x);
+}
```

Comment:
This is confusing.  These methods do different things based on what type is passed to them, but that's a little weird since a MarkDownEmphasisHtmlGenerator *is* a MarkDownHtmlGenerator.  When i looked at the calling code in TryCapturingRightEmphasisToken it looked like you were just appending the same thing twice.  Let's give these methods different names since they do different things (for example: AppendToTokens and AppendToLookUpTable).  That should also cut down on the amount of casting you need to do.